bcos-ledger/src/libledger/Ledger.cpp

```diff
@@ -1494,17 +1495,36 @@ void Ledger::getTxProof(
                             auto merkleProofPtr = std::make_shared<MerkleProof>();
                             auto anyHasher = cryptoSuite->hashImpl()->hasher();
                             std::visit(
-                                [txHash = std::move(_txHash), &_txList, &merkleProofPtr](
-                                    auto&& hasher) {
+                                [this, txHash = std::move(_txHash), &_txList, &merkleProofPtr,
+                                    &blockNumber](auto&& hasher) {
                                     using Hasher = std::remove_reference_t<decltype(hasher)>;
                                     bcos::crypto::merkle::Merkle<Hasher> merkle;
                                     auto hashesRange =
                                         _txList | RANGES::views::transform(
                                                       [](const Transaction::Ptr& transaction) {
                                                           return transaction->hash();
                                                       });
+
+                                    std::shared_ptr<std::vector<h256>> merkleTree;
+                                    {
+                                        UniqueGuard l(m_txMerkleMtx);
+                                        // Get merkle from m_txProofMerkleCache
+                                        auto& cacheRef = m_txProofMerkleCache[blockNumber].value();
```

Comment:
这里锁的范围可以考虑缩小一些，m_txProofMerkleCache 用了tbb的并行容器，支持并发访问。

```suggestion
                                     auto merkleTree = m_txProofMerkleCache[blockNumber].value();
                                     if (!merkleTree)
                                    {
                                        UniqueGuard l(m_txMerkleMtx);
                                        // Get merkle from m_txProofMerkleCache
                                       cacheRef = m_txProofMerkleCache[blockNumber].value();
```